lib/lz4.c

```diff
@@ -446,9 +447,9 @@ static const U32 LZ4_skipTrigger = 6;  /* Increase this value ==> compression ru
 *  Local Structures and types
 **************************************/
 typedef enum { notLimited = 0, limitedOutput = 1 } limitedOutput_directive;
-typedef enum { byPtr, byU32, byU16 } tableType_t;
+typedef enum { unusedTable = 0, byPtr = 1, byU32 = 2, byU16 = 3 } tableType_t;
 
-typedef enum { noDict = 0, withPrefix64k, usingExtDict } dict_directive;
+typedef enum { noDict = 0, withPrefix64k, usingExtDict, usingExtDictCtx } dict_directive;
```

Comment:
Wording of the last 2 directives feel a bit too close, which introduces some confusion.

My understanding is that your patch introduces a new member within compression context, called `cctx->dictCtx`. It's itself a pointer to another compression context. When the new directive is set, all information related to external dictionary must be read from `dictCtx`.

I was expecting that, for familiarity with `zstd` API, you would use something like `CDict`. It's true it's not a separate type though, so maybe `dictCtx` is more appropriate.
In which case, `usingDictCtx` feel like a better name.

Also : try commenting what each directive means / does.